src/Spice86.Core/Emulator/CPU/CfgCpu/ParsedInstruction/Discriminator.cs

```diff
@@ -0,0 +1,130 @@
+namespace Spice86.Core.Emulator.CPU.CfgCpu.ParsedInstruction;
+
+using System.Collections.Immutable;
+
+/// <summary>
+/// Base class for objects that have a discriminator value based on a list of bytes.
+/// The bytes are used to compare 2 discriminators.
+/// If at the same position bytes are equal, the discriminators are considered equals.
+/// If there are null bytes they are not considered for the equality comparison of the position
+/// If the length of the discriminator value differs, discriminators are different
+/// </summary>
+public class Discriminator : IComparable<Discriminator> {
+    public Discriminator(ImmutableList<byte?> discriminatorValue) {
+        DiscriminatorValue = discriminatorValue;
+    }
+
+    /// <summary>
+    /// Value of the discriminator
+    /// </summary>
+    public ImmutableList<byte?> DiscriminatorValue { get; }
```

Comment:
This can be a FrozenSet. Frozen collections are truly immutable, and the fastest for reading.